### DISCUSSION TITLE
Can we close sync_service server without closing the listener first

### DISCUSSION THREAD

**Discussion by apodgorbunschih** [open-feature/flagd#1373] [Q&A] [answered]
Hello People,

I am using FlagD in a K8S cluster, and I see a lot of warning like `error from sync server start: accept tcp [::]:8015: use of closed network connection`

When debugging I found that the `shutdown` is closing first the `listener` then the `server` 

https://github.com/open-feature/flagd/blob/ae2459504f7eccafebccec83fa1f72b08f41a978/flagd/pkg/service/flag-sync/sync_service.go#L121-L127

When the `listener` is closed the server is now having a closed connection that is leading to the above warnings.

Is it possible to `stop` the server without closing the `listener` ?

I see that in the `s.server.Stop()` is a calling `s.closeListenersLocked()` but not sure if its different from `s.listener.Close()`

This is the testing scenario.
![Screenshot 2024-08-01 at 08 30 04](https://github.com/user-attachments/assets/9654343c-c642-4843-b5ed-47e877323fb3)

**Comment by Kavindu-Dodan** [comment, marked as answer]
@apodgorbunschih thanks for reporting this behavior. Yes, `s.server.Stop()` (grpc server stop) already close the internal listener. I opened https://github.com/open-feature/flagd/pull/1375 to improve this

**Reply by apodgorbunschih** [reply]
@Kavindu-Dodan Thanks a lot for such a quick fix ❤️
